src/lib/y2network/interface.rb

```diff
@@ -16,17 +16,50 @@
 #
 # To contact SUSE LLC about this file by physical or electronic mail, you may
 # find current contact information at www.suse.com.
+
+require "y2network/interface_type"
+require "y2network/hwinfo"
+
 module Y2Network
   # Network interface.
   class Interface
-    # @return [String] Device name (eth0, wlan0, etc.)
+    # @return [String] Device name ('eth0', 'wlan0', etc.)
     attr_accessor :name
+    # @return [String] Interface description
+    attr_accessor :description
+    # @return [Symbol] Interface type
+    attr_accessor :type
+    attr_reader :configured
```

Comment:
```suggestion
    # @return [Boolean,nil]
    attr_reader :configured
```

`init` allows a nil value, which I think is really a minor type bug